pkg/controllers/data.go

```diff
@@ -82,7 +84,29 @@ func (d *DataController) Save(w http.ResponseWriter, r *http.Request) {
 	}
 	respondWithJson(w, http.StatusCreated, thing)
 	defer r.Body.Close()
+}
+
+func (d *DataController) DeleteByDeviceID(w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	deviceId := params["deviceId"]
+
+	err := d.DataInteractor.DeleteByDeviceID(deviceId)
+	if err != nil {
+		respondWithError(w, http.StatusBadGateway, "Invalid information")
```

Comment:
I think this error handling can be improved. The correct approach is to return the error according to the right protocol semantics as can be seen [here](https://tools.ietf.org/html/rfc7231#section-6.5). For example, when the client does the request with a wrong payload it's common to return the [`422 - Unprocessable entity`](https://evertpot.com/http/422-unprocessable-entity) status code.

Reply:
Done!